### PR TITLE
Pull images before running CI

### PR DESCRIPTION
**Pull request by vdemeester**
The idea is simple : when running the CI, the integration step should not pull anything 👼

### CODE REVIEW

**Comment by vdemeester** [comment]
Removing `circle.yml` because we don't use it anymore.

File: .travis.yml [hunk 1]
```diff
@@ -23,12 +23,11 @@ before_install:
...
   - docker version
   - pip install --user -r requirements.txt
+  - make pull
 before_script:
   - make validate
   - make binary
-script:
-  - travis_retry make test-unit
-  - travis_retry make test-integration
+script: travis_retry make test-unit && travis_retry make test-integration
```
Comment (vdemeester) [inline .travis.yml]: This changes is simple : **if test-unit fails, DON'T run integration test** 👼.

Comment (timoreimann) [inline .travis.yml]: Should we remove the retry on the unit test execution right away?

File: Makefile [hunk 2]
```diff
@@ -82,5 +82,11 @@ lint:
 
 fmt:
 	gofmt -s -l -w $(SRCS)
+
+pull:
```
Comment (timoreimann) [inline Makefile]: On second thought: can we call this target `pull-images`?

`pull` seems a bit too generic / unspecific to me.

**Comment by SantoDE** [comment]
LGTM once tests are green 👼

File: Makefile (hunk 2 above)
Comment (vdemeester) [inline Makefile]: I do agree 👍

File: .travis.yml (hunk 1 above)
Comment (vdemeester) [inline .travis.yml]: meh, I want to fix the flakeys before 😛

Comment (timoreimann) [inline .travis.yml]: :+1:

**Comment by timoreimann** [comment]
Relates to #513.

**Pull request merged**